商品情報編集機能

wer-jp-kyaw-min-hein opened the pull request:
# What

このプルリクエストでは、商品情報編集機能を実装しました。この機能により、ユーザーはシステム内の商品情報を編集することができます。

# Why

以下の理由から、この機能の実装は重要です：

1. **ユーザーエクスペリエンスの向上**：ユーザーが商品情報を編集できるようにすることで、より制御と柔軟性を提供します。これにより、プラットフォーム全体のユーザーエクスペリエンスと満足度が向上します。

2. **正確で最新の情報**：商品情報の編集機能により、出品者はリスティングの正確性と最新性を保つことができます。これにより、誤解を招く古い情報を避け、信頼性のあるマーケットプレイスを形成します。

3. **ユーザーリクエストへの対応**：ユーザーからは、初期リスティング後に商品の詳細を変更する必要があるとの要望がありました。この機能の実装により、ユーザーリクエストに応えることができます。

wer-jp-kyaw-min-hein committed:
アイテム編集

index(path)入れ/def edit生産/edit(form_with)model名入れ/items_controllerにdef updateいれ

wer-jp-kyaw-min-hein committed:
redirection

div-reviewer reviewed (commented):
Leebal13様、学習お疲れ様です！
商品編集の初回レビュー依頼ありがとうございます。プルリクエスト確認させていただきました。

共同動画でも出品者かどうかわかりやすいようにカーソルを動かしたりと読み手を考えた作業ができており大変素晴らしいと思いました。
修正点をコメントさせて頂きましたので、ご確認をお願い致します。

【次回プルリクエスト提出する際の注意点】
```ruby
①修正が終わりましたら、`コミットとプッシュ`をして、`プルリクエストに修正内容が反映されているか確認`しましょう。
②既に頂いている挙動確認の画像や動画が修正範囲だった場合、最新の画像や動画の提出をお願い致します。
③googleフォームの送信をお願いいたします。
```

**ご不明点や疑問点ございましたら、チャット・コネクトメンターにご相談ください！**
どうぞよろしくおねがいします。

wer-jp-kyaw-min-hein committed:
修正

set up before action with find method
make path to 戻る button
入れedit in authenticate_user!

div-reviewer reviewed (commented):
Leebal13様、学習お疲れ様です！
商品編集の再レビュー依頼ありがとうございます。プルリクエスト確認させていただきました。

段々とコードが良くなってきていますね。この調子で頑張りましょう。
修正点をコメントさせて頂きましたので、ご確認をお願い致します。

【次回プルリクエスト提出する際の注意点】
```ruby
①修正が終わりましたら、`コミットとプッシュ`をして、`プルリクエストに修正内容が反映されているか確認`しましょう。
②既に頂いている挙動確認の画像や動画が修正範囲だった場合、最新の画像や動画の提出をお願い致します。
③googleフォームの送信をお願いいたします。
```

**ご不明点や疑問点ございましたら、チャット・コネクトメンターにご相談ください！**
どうぞよろしくおねがいします。

wer-jp-kyaw-min-hein committed:
@item = Item.find(params[:item_id])消し

div-reviewer reviewed (commented):
Leebal13様、学習お疲れさまです。
レビュー依頼ありがとうございます！

前回のコメントにあった修正点が綺麗に直せましたね。素晴らしいです。
こちら`LGTM`とさせていただきます！実装お疲れさまでした。

【注意】
```ruby
①（次のブランチを切る前に）プルリクエストのMergeとmainブランチへのPullを忘れないようにしましょう。
②次のフェーズに進む際は必ず新しいブランチを作ってから作業しましょう。
```

ご不明点や疑問点ございましたら、チャット・コネクトメンターにご相談ください！
次は**「商品削除機能」**かと思われます！引き続き頑張りましょう！応援しております！
